src/lib.rs

```diff
@@ -130,36 +130,24 @@ use doc_comment::doctest;
 doctest!("../Readme.md");
 
 // Check that the type of the expression passed here implements IntoIterator.
-// Hopefully this optimizes away in release builds.
 #[doc(hidden)]
-#[macro_export]
-macro_rules! assert_impl_into_iter {
-    ($x: expr) => {
-        let _ = || {
-            fn assert_impl_into_iter<T>(_: T)
-            where
-                T: IntoIterator,
-            {
-            }
-            assert_impl_into_iter($x);
-        };
-    };
-}
+#[inline(always)]
```

Comment:
Is this necessary? I don't know how aggressively the compiler optimizes by default, but is an empty function not something that is always optimized away?

Reply:
Functions aren't inlined across crates unless either specifically hinted or LTO is enabled. Here, we definitely know here that generating a function call will be more expensive than inlining this.

Reply:
Oh, that's a good thing to know about.
Let's keep it then

Reply:
Oh, that's a good thing to know about.
Let's keep it then